### DISCUSSION TITLE
How do I create a new temporary window from the LSP server?

### DISCUSSION THREAD

**Discussion by RomanMIzulin** [microsoft/vscode-discussions#880] [Extension Development QnA] [answered]
I generate some code and want to place it in a new split window. Like Copilot generates 10 suggestions.
How can I achieve this behavior?  
```
@LSP_SERVER.command("unittest_generator.run")
def generate_test_template(params: lsp.ExecuteCommandParams) -> None:
    lsp. # some command here?
```

**Comment by isidorn** [comment]
What exactly are you trying to achieve here? What is your extension doing? Thanks

**Reply by RomanMIzulin** [reply]
Extension should generate Python unit tests templates into separate tab(ready for copy/paste).
Here my objective is to capture command from vscode and get reference to the code on which the command was called, generate a related template for tests and put into new separate window. 
 
Here, Copilot has opened a new tab. 
![image](https://github.com/microsoft/vscode-discussions/assets/10415843/25641a38-7a6d-4c2b-ae4f-dd656d66d6a2)

**Reply by isidorn** [reply, marked as answer]
Well you can try to use the Editor API to open a new custom editor on the side and control it's content. I suggest to explore our API and sampleshttps://github.com/microsoft/vscode-extension-samples
